docs/docs/file-system-route-api.md

```diff
@@ -0,0 +1,299 @@
+
+Gatsby also supports _splat_ routes, which are routes that will match _anything_ after the splat. These are less common, but still have use cases. As an example, suppose that you are rendering images from [S3](/docs/deploying-to-s3-cloudfront/) and the URL is actually the key to the asset in AWS. Here is how you might create your file:
+
+- `src/pages/image/[...awsKey].js => /image/*awsKey`
+- `src/pages/image/[...].js => /image/*`
+
+Three periods `...` mark a page as a splat route. Optionally, you can name the splat as well, which has the benefit of naming the key of the property that your component receives. The dynamic segment of the file name (the part between the square brackets) will be filled in and provided to your components on a `props.params` object. For example:
+
+```js:title=src/pages/users/[id].js
+function UserPage(props) {
+  const id = props.params.id
+}
+```
+
+```js:title=src/pages/image/[...awsKey].js
+function ProductsPage(props) {
+  const splat = props.params.awsKey
+}
+```
+
+```js:title=src/pages/image/[...].js
+function AppPage(props) {
+  const splat = props.params[‘*’]
+}
+```
+
+## Routing and linking
+
+Gatsby "slugifies" every route that gets created from collection pages. When you want to link to one of those pages, it may not always be clear how to construct the URL from scratch.
+
+To address this issue, Gatsby automatically includes a `gatsbyPath` field on every type used by collection pages. The `gatsbyPath` field must take an argument of the `filePath` it is trying to resolve. This is necessary because it’s possible that one type is used in multiple collection pages.
+
+There are some general syntax requirements when using the `filePath` argument:
+
+- The path must be an absolute path (so starting with a `/`)
+- You must omit the file extension
+- You must omit the `src/pages` prefix
+- Your path must not include `index`
+
+### `gatsbyPath` example
+
+Assume that a `Product` type is used in two pages:
+
+- `src/pages/products/{Product.name}.js`
+- `src/pages/discounts/{Product.name}.js`
+
+If you wanted to link to the `products/{Product.name}` and `discounts/{Product.name}` routes from your home page, you would have a component like this:
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { Link, graphql } from "gatsby"
+
+export default function HomePage(props) {
+  return (
+    <ul>
+      {props.data.allProduct.map(product => (
+        <li key={product.name}>
+          <Link to={product.productPath}>{product.name}</Link> (<Link to={product.discountPath}>Discount</Link>)
+        </li>
+      ))}
+    </ul>
+  )
+}
+
+export const query = graphql`
+  query {
+    allProduct {
+      name
+      productPath: gatsbyPath(filePath: "/products/{Product.name}")
+      discountPath: gatsbyPath(filePath: "/discounts/{Product.name}")
+    }
+  }
+}
+```
+
+By using [aliasing](/docs/graphql-reference/#aliasing) you can use `gatsbyPath` multiple times.
+
+## Example use cases
```

Comment:
In the [future structure of the docs](https://documentation.divio.com/), do you think these example use cases could stand on their own as separate How-To Guides, or would you recommend keeping them in this Reference doc? (Asking so I can make a note in my spreadsheet :) )

Reply:
I'd put them to "How-To Guides" 👍 💯